docs/reference/data-streams/modify-data-streams-api.asciidoc

```diff
@@ -0,0 +1,112 @@
+[[modify-data-streams-api]]
+=== Modify data streams API
+++++
+<titleabbrev>Modify data streams API</titleabbrev>
+++++
+
+Performs one or more data stream modification actions in a single atomic
+operation.
+
+////
+[source,console]
+----
+PUT /_index_template/template
+{
+  "index_patterns": ["my-logs*"],
+  "data_stream": { }
+}
+
+POST my-logs/_doc/
+{
+    "message" : "trying out DS modification",
+    "@timestamp": "2021-01-01"
+}
+
+POST my-logs/_rollover
+
+POST index-to-add/_doc/
+{
+    "description" : "this is the index to be added to a data stream",
+    "@timestamp": "2021-01-01"
+}
+----
+////
+
+[source,console]
+----
+POST _data_stream/_modify
+{
+  "actions": [
+    {
+      "remove_backing_index": {
+        "data_stream": "my-logs",
+        "index": ".ds-my-logs-2021.10.15-000001"
+      }
+    },
+    {
+      "add_backing_index": {
+        "data_stream": "my-logs",
+        "index": "index-to-add"
+      }
+    }
+  ]
+}
+----
+// TEST[continued]
+
+////
+[source,console]
+-----------------------------------
+DELETE /_data_stream/my-logs
+DELETE /_index_template/template
+-----------------------------------
+// TEST[continued]
+////
+
+[[modify-data-streams-api-request]]
+==== {api-request-title}
+
+`POST /_data_stream/_modify`
+
+[role="child_attributes"]
+[[modify-data-streams-api-request-body]]
+==== {api-request-body-title}
+
+`actions`::
+(Required, array of objects) Actions to perform.
++
+.Properties of `actions` objects
+[%collapsible%open]
+====
+`<action>`::
+(Required, object) The key is the action type. At least one action is required.
++
+.Valid `<action>` keys
+[%collapsible%open]
+=====
+`add_backing_index`::
+Adds an existing index as backing index to a data stream. The index will be
+hidden as part of the operation.
+
+WARNING: Addition of certain indices with the `add_backing_index` operation
+can result in improper data stream behavior. This is a potentially
+destructive action and should be considered an expert level API.
```

Comment:
Can you be more specific about what indices cause problems? Otherwise, I'd just drop the qualification. (It sounds scarier, but that's probably not a bad thing.)

```suggestion
WARNING: Adding indices with the `add_backing_index` action
can potentially result in improper data stream behavior. 
This should be considered an expert level API.
```